ledger/src/blockstore_meta.rs

```diff
@@ -138,6 +138,16 @@ pub(crate) struct ErasureConfig {
     num_coding: usize,
 }
 
+#[derive(Clone, Copy, Debug, Eq, PartialEq, Serialize, Deserialize)]
+pub struct MerkleRootMeta {
+    /// The merkle root
+    merkle_root: Hash,
+    /// The first received shred index
+    first_received_shred_index: u64,
```

Comment:
shred indices are u32:
https://github.com/solana-labs/solana/blob/a9509f56b/ledger/src/shred.rs#L206
any reason to use u64 here?

Reply:
I used `u64` here because although indices are `u32`, the `ShredCode` and `ShredData` are keyed with `u64`. I followed the same style as `ErasureMeta::first_coding_index`, which uses `u64`.

If you prefer I can store it as a `u32`, and perform the cast when doing the blockstore read

Reply:
I don't know any benefit to using `u64` here.
the source of truth is the `Shred` structure, so I lean to keep things consistent with `Shred`.

Reply:
I'm onboard with `u32` here as well; it is kind of annoying that the columns Ashwin mentioned that do use `u64` (like the shred) columns do the `u64 <==> u32` conversion on insert/fetch